tests/test_flags/test_dry_run.py

```
@@ -0,0 +1,524 @@
+                except SystemExit:
+                    pass
+
+                # Should exit with git's failure code
+                mock_exit.assert_called_with(1)
+
+    def test_main_flow_with_dry_run(self):
+        """Test the main flow with --dry-run flag."""
+        test_argv = ["git-commitai", "--dry-run"]
+
+        with patch("sys.argv", test_argv):
+            with patch("subprocess.run") as mock_subprocess:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)